pkg/plugins/golang/declarative/v1/scaffolds/internal/templates/types.go

```diff
@@ -33,18 +33,30 @@ type Types struct {
 	machinery.MultiGroupMixin
 	machinery.BoilerplateMixin
 	machinery.ResourceMixin
+
+	IsLegacyLayout bool
 }
 
 // SetTemplateDefaults implements file.Template
 func (f *Types) SetTemplateDefaults() error {
 	if f.Path == "" {
-		if f.MultiGroup {
-			f.Path = filepath.Join("apis", "%[group]", "%[version]", "%[kind]_types.go")
+		if f.IsLegacyLayout {
+			if f.MultiGroup {
+				f.Path = filepath.Join("apis", "%[group]", "%[version]", "%[kind]_types.go")
+			} else {
+				f.Path = filepath.Join("api", "%[version]", "%[kind]_types.go")
+			}
 		} else {
-			f.Path = filepath.Join("api", "%[version]", "%[kind]_types.go")
+			if f.MultiGroup {
+				f.Path = filepath.Join("pkg", "apis", "%[group]", "%[version]", "%[kind]_types.go")
+			} else {
+				f.Path = filepath.Join("pkg", "apis", "%[version]", "%[kind]_types.go")
```

Comment:
will it be `/pkg/apis` and not `/pkg/api` even if it is not MultiGroup?

Reply:
yes, that sort outs the old bug. That should never be api and apis at the same way that controllers is only controllers in both scenarios. 

See that make no sense we have api in the singular (the dir can have 1 to many independently ) and it brings a lot of complexities which was reduced in this PR for the new layout. It is something described in the migration doc as well.

Reply:
Closing this one as sorted out @umangachapagain,
But feel free to re-open or ping me if you need.